janus_server/src/datastore.rs

```diff
@@ -2169,6 +2330,94 @@ mod tests {
         assert!(rows.len() == 2);
     }
 
+    #[tokio::test]
+    async fn roundtrip_batch_aggregation() {
+        install_test_trace_subscriber();
+
+        let task_id = TaskId::random();
+        let other_task_id = TaskId::random();
+        //let vdaf = Prio3Aes128Count::new(2).unwrap();
+        let aggregate_share = AggregateShare::from(vec![Field64::from(17)]);
+
+        let (ds, _db_handle) = ephemeral_datastore().await;
+
+        let batch_aggregations: Vec<BatchAggregation<AggregateShare<Field64>>> = ds
+            .run_tx(|tx| {
+                let aggregate_share = aggregate_share.clone();
+                Box::pin(async move {
+                    tx.put_task(&new_dummy_task(
+                        task_id,
+                        Vdaf::Prio3Aes128Count,
+                        Role::Leader,
+                    ))
+                    .await?;
+
+                    tx.put_task(&new_dummy_task(
+                        other_task_id,
+                        Vdaf::Prio3Aes128Count,
+                        Role::Leader,
+                    ))
+                    .await?;
+
+                    tx.put_batch_aggregation(task_id, Time(100), &aggregate_share, 0, &[0; 32])
+                        .await?;
+
+                    tx.put_batch_aggregation(task_id, Time(200), &aggregate_share, 0, &[0; 32])
+                        .await?;
+
+                    tx.put_batch_aggregation(task_id, Time(300), &aggregate_share, 0, &[0; 32])
+                        .await?;
+
+                    tx.put_batch_aggregation(
+                        other_task_id,
+                        Time(200),
+                        &aggregate_share,
+                        0,
+                        &[0; 32],
+                    )
+                    .await?;
+
+                    tx.increment_batch_aggregation_times_collected(task_id, Time(200))
+                        .await?;
+
+                    tx.increment_batch_aggregation_times_collected(task_id, Time(200))
+                        .await?;
+
+                    tx.get_batch_aggregations(
+                        task_id,
+                        Interval {
+                            start: Time(50),
+                            duration: Duration(200),
+                        },
+                    )
+                    .await
+                })
+            })
+            .await
+            .unwrap();
+
+        let expected_batch_aggregations = [
```

Comment:
pedantic nitpick: I think this is currently ordered, so if Postgres changes to return the rows in opposite order, this test will break. Drop ordering on the check in this test?